sklearn/linear_model/_glm/_newton_solver.py

```diff
@@ -516,3 +532,460 @@ def inner_solve(self, X, y, sample_weight):
                 )
             self.use_fallback_lbfgs_solve = True
             return
+
+
+class NewtonLSMRSolver(NewtonSolver):
```

Comment:
I have the feeling the code would be simpler to follow if we had a special subclass to handle the `if self.linear_loss.base_loss.is_multiclass` case.

That would require introducing a factory function to do the dispatching to the correct solver class based on the loss object but I have the feeling that would be worth it.